include/dt-bindings/timers/stm32_timers_h7.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2020 Teslabs Engineering S.L.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_TIMERS_STM32_TIMERS_H7_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_TIMERS_STM32_TIMERS_H7_
+
+/* Helper macro to generate ITR (for master/slave in timers bits 4 and 5 of
+ * SMCR register).
+ */
+#define DT_STM32_TIMERS_ITR(n)		((n & 0x3U) << 4U)
```

Comment:
This formula is valid only when ITR number is lower or equal to 3. 
But other series like STM32G4 can have ITR number up to 10.
And unfortunately, all bits of ITR are not adjacent one from each other (Bits 21, 20, 6, 5, 4 TS[4:0]: Trigger selection (see bits 21:20 for TS[4:3])).
I will discuss with @erwango on how to handle this issue with best genericity.

Reply:
This header only applies to H7, where ITR goes from 0..3, so bits are adjacent. If some other series use non-adjacent bits the formula needs to take that into account.

Reply:
@ABOSTM After looking at G4 series, I think that the idea of having a macro that converts a number to TS is not a good idea, as trigger selection number does not match the actual TS number. In case of G4, datasheet is confusing for (01000 ... 01110).

![image](https://user-images.githubusercontent.com/25011557/87654488-414f0000-c757-11ea-8b0c-e4c634623bf5.png)

Reply:
@ABOSTM can you check latest changes wrt trigger selection in DT?